meson.build

```diff
@@ -0,0 +1,53 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+project('debugcc',
+  'c',
+  license: [ 'BSD-3-Clause' ],
+  meson_version : '>= 0.61.0', # for install_symlink
+  default_options: [ 'buildtype=release',
+    ])
+
+platforms = [
+  'msm8936',
+  'msm8994',
+  'msm8996',
+  'msm8998',
+  'qcs404',
+  'sc8280xp',
+  'sdm845',
+  'sm6115',
+  'sm6125',
+  'sm6350',
+  'sm6375',
+  'sm8150',
+  'sm8250',
+  'sm8350',
+  'sm8450',
+  'sm8550',
+  ]
+
+debugcc_srcs = [
+  'debugcc.c',
+  ]
+
+platform_strings = []
+
+foreach p: platforms
+  debugcc_srcs += p + '.c'
+  platform_strings += 'extern struct debugcc_platform ' + p + '_debugcc;'
+
+  install_symlink(p + '-debugcc',
+    install_dir: get_option('bindir'),
+    pointing_to: 'debugcc')
+endforeach
+
+debugcc_srcs += configure_file(
+  output: 'platforms.h',
+  capture: true,
+  command: ['echo',
+    '/* Autogenerated file, do not edit */\n\n' +
+    '\n'.join(platform_strings)])
```

Comment:
Is it just me who thinks this is really dirty? I love the idea of autogenerating the file, but we have the string contents in a Meson variable... And have to run that _through a shell command_ to get it into the file?!? Doesn't `configure_file()` have a `file_contents: str` sort of argument for this use-case?

Reply:
Maybe it actually makes sense to create a platforms.h.in file with the static contents (comment header, static inline for the platforms and use @NAME@ syntax via meson variables to fill in the contentsñ

Reply:
Unfortunately no. I think this is the best way I have for now. Suggestions are appreciated.

Reply:
Sure, I'll see if I can submit a PR with that idea later, if it turns out to work.